tests/resources/transaction/test_pushtx.py

```diff
@@ -21,6 +21,9 @@ class BasePushTxTest(_BaseResourceTest._ResourceTest):
 
     is_post: Optional[bool] = None
 
+    # XXX: we will get a "two instances of the same tx in memory" otherwise
```

Comment:
It's basically because of `test_spending_voided`, it'll happen here:

```python
          # Now we set this tx2 as voided and try to push a tx3 that spends tx2
          tx_meta = tx2.get_metadata()
          assert tx2.hash is not None
          tx_meta.voided_by = {tx2.hash}
>         self.manager.tx_storage.save_transaction(tx2, only_metadata=True)   
```

I'm not entirely sure whether this is a bug on how RocksDB forwards the `save_transaction` method, it calls `self._save_to_weakref(tx)` even if we're only saving the metadata.

Since this is only because of how we specifically build the tests I'd rather not change any of that right now and deal with it later.